iron-input.html

```diff
@@ -12,300 +12,386 @@
+          input.dispatchEvent(new Event('feature-check-dispatch-event'));
+        } catch(e) {}
+
+        return canDispatch;
       },
 
       /**
-       * Set to true to prevent the user from entering invalid input. If `allowedPattern` is set,
-       * any character typed by the user will be matched against that pattern, and rejected if it's not a match.
-       * Pasted input will have each character checked individually; if any character
-       * doesn't match `allowedPattern`, the entire pasted string will be rejected.
-       * If `allowedPattern` is not set, it will use the `type` attribute (only supported for `type=number`).
+       * @this {Node}
+       * @param {!Event} event
+       * @return {boolean}
        */
-      preventInvalidInput: {
-        type: Boolean
+      _dispatchEventFirefoxIE: function(event) {
+        // Due to Firefox bug, events fired on disabled form controls can throw
+        // errors; furthermore, neither IE nor Firefox will actually dispatch
+        // events from disabled form controls; as such, we toggle disable around
+        // the dispatch to allow notifying properties to notify
+        // See issue #47 for details
+        var disabled = this.disabled;
+        this.disabled = false;
+        var defaultPrevented = this._origDispatchEvent(event);
+        this.disabled = disabled;
+        return defaultPrevented
+      },
+
+      get _patternRegExp() {
+        var pattern;
+        if (this.allowedPattern) {
+          pattern = new RegExp(this.allowedPattern);
+        } else {
+          switch (this.type) {
+            case 'number':
+              pattern = /[0-9.,e-]/;
+              break;
+          }
+        }
+        return pattern;
       },
 
       /**
-       * Regular expression that list the characters allowed as input.
-       * This pattern represents the allowed characters for the field; as the user inputs text,
-       * each individual character will be checked against the pattern (rather than checking
-       * the entire value as a whole). The recommended format should be a list of allowed characters;
-       * for example, `[a-zA-Z0-9.+-!;:]`
+       * @suppress {checkTypes}
        */
-      allowedPattern: {
-        type: String,
-        observer: "_allowedPatternChanged"
-      },
+      _bindValueChanged: function(bindValue, inputElement) {
+        // The observer could have run before attached() when we have actually initialized
+        // this property.
+        if (!inputElement)
+          return;
+
+        if (bindValue === undefined) {
+          inputElement.value = null;
+        } else {
+          inputElement.value = bindValue;
+        }
 
-      _previousValidInput: {
-        type: String,
-        value: ''
+        if (this.autoValidate) {
+          this.validate();
+        }
+
+        // manually notify because we don't want to notify until after setting value
+        this.fire('bind-value-changed', {value: bindValue});
```

Comment:
Yes, I think it does (there were offline discussions about it I don't recall 😭 )